src/app/util/attribute-storage.cpp

```diff
@@ -1122,6 +1122,10 @@ void emAfLoadAttributeDefaults(EndpointId endpoint, bool ignoreStorage)
     uint16_t epCount = emberAfEndpointCount();
     uint8_t attrData[ATTRIBUTE_LARGEST];
     auto * attrStorage = ignoreStorage ? nullptr : app::GetAttributePersistenceProvider();
+    if (!ignoreStorage && (attrStorage == nullptr))
+    {
+        ChipLogProgress(DataManagement, "Attribute persistence needs a persistence provider");
```

Comment:
Suggest then having:
```
VerifyOrDie(!ignoreStorage && (attrStorage == nullptr) && "Attribute persistence needs a persistence provider");
```

Reply:
Done.

Reply:
But I put this down where we actually have an NVM attribute, because we have various configurations, including clients that have no attributes at all, that would otherwise need to implement a dummy storage provider.